src/Service.Tests/CosmosTests/QueryFilterTests.cs

```diff
@@ -970,7 +971,8 @@ public async Task TestQueryFilterNestedFieldAuth_AuthorizedNestedField()
                 }
             }";
 
-            JsonElement actual = await ExecuteGraphQLRequestAsync(_graphQLQueryName, query: gqlQuery);
+            string authToken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: AuthorizationType.Authenticated.ToString());
+            JsonElement actual = await ExecuteGraphQLRequestAsync(_graphQLQueryName, query: gqlQuery, authToken: authToken);
```

Comment:
by changing the role from anonymous to `authenticated` you're changing the behavior of the test. The test description looks to be describing this test as validating behavior for the anonymous role.

Reply:
I changed anonymous to `limited-read-role`. I think due to positioning, comparison is coming like this. `authenticated` role is as it is.